TelepresenceBot/server-unit/core/server.js

```diff
@@ -1,21 +1,64 @@
 var io = require('socket.io').listen(5000);
+var ClientType = require("./clientType.js");
+var LoggingClient = require("./loggingClient.js");
 
-var clients = {};
+var room = "London";
+var humans = [];
+var bots = [];
+var testClient = new LoggingClient();
+
+io.use(function(client, next){
```

Comment:
This function establishes whether the client is allowed to connect to the server.

A `Human` client is allowed to connect when a bot is present in the `London` room and no other `Human` client is present in the same room.

A `Bot` and a `Test` client is always allowed to connect regardless.